lib/ios-crash-log.js

```diff
@@ -0,0 +1,61 @@
+import path from 'path';
+import _ from 'lodash';
+import _glob from 'glob';
+import { fs, getLogger } from 'appium-support';
+import B from 'bluebird';
+import { asyncmap } from 'asyncbox';
+
+const glob = B.promisify(_glob);
+
+
+require('date-utils');
+const logger = getLogger('IOS-crash-log');
+const CRASH_DIR = path.resolve(process.env.HOME, 'Library', 'Logs', 'DiagnosticReports');
+
+class IOSCrashLog { 
+  constructor() {
+    this.prevLogs = [];
+    this.logsSinceLastRequest = [];
+  }
+
+  async getCrashes () {
+    try {
+      return await glob(`${CRASH_DIR} /*.crash`);
+    } catch (err) {
+      logger.errorAndThrow(`There was a problem getting the crash list: ${err}`);
+      throw err;
+    }
+  }
+
+  async filesToJSON (files) {
+    return asyncmap(files, async (f) => {
+      let filename = path.resolve(this.crashDir, f);
+      let stat = await fs.stat(filename);
+      return {
+        timestamp: stat.ctime,
+        level: 'ALL', 
+        message: await fs.readFile(filename, 'utf8')
+      };
+    });
+  }
+
+  async startCapture () {
+    let fileList = await this.getCrashes();
+    this.prevLogs = fileList;
+  }
+
+  async getLogs () {
+    let crashFiles = await glob(`${CRASH_DIR} /*.crash`);
```

Comment:
Do we really want a space in this path? It looks weird. If it **is** supposed to be there, a little note might be in order so that it doesn't catch the eye.